pgrx/src/array.rs

```diff
@@ -373,10 +374,63 @@ impl RawArray {
         let ptr = self.ptr.as_ptr().cast::<u8>();
         ptr.wrapping_add(unsafe { varlena::varsize_any(ptr.cast()) })
     }
+
+    /// Slightly faster than new_array_type_with_len(0)
+    pub fn new_empty_array_type<T>() -> Result<RawArray, ArrayAllocError>
+    where
+        T: ArrayFastAllocSubType,
```

Comment:
This can just use the `Scalar` trait bound. Hmm. I should probably move that into `pgrx::array` I guess, since it will outlast `datum::Array`.